frontend/src/search/search.js

```diff
@@ -0,0 +1,315 @@
+                                        return (
+                                            <option value={valueArray} key={key}>
+                                                {photographer.name} [Unknown]
+                                            </option>
+                                        );
+                                    }
+                                    return (
+                                        <option value={valueArray} key={key}>
+                                            {photographer.name} [{photographer.number}]
+                                        </option>
+                                    );
+                                })
+                            }
+                        </select>
+                    </label>
+                    <br/>
+                    <label className='input-div'>
+                        <p className='search-label'>Caption:</p>
+                        <input
+                            className='search-text-input'
+                            type="text"
+                            name="caption"
+                            value={this.state.caption}
+                            onChange={this.handleChange}
+                        />
+                    </label>
+                    <br/>
+                    <label className='input-div'>
+                        <p className='search-label'>Tags:</p>
+                        <Select
+                            defaultValue={this.state.tags}
+                            isMulti
+                            name="tags"
+                            options={tagOptions}
+                            onChange={this.handleSelectDropdownChange}
+                            menuPlacement="auto"
+                            menuPosition="fixed"
+                        />
+                    </label>
+                    <br/>
+                    <input type="submit" value="Search" />
+                </form>
+            </div>
+        );
+    }
+}
+SearchForm.propTypes = {
+    updateSearchData: PropTypes.func,
+    tagData: PropTypes.array,
+    photographerData: PropTypes.array,
+};
+
+export class Search extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            loading: true,
+            data: null,
+            isAdvanced: false,
+            searchedText: '',
+            tagData: null,
+            photographerData: null,
+        };
+    }
+
+    updateSearchData = (searchData) => {
+        this.setState({ ...searchData });
+    }
+
+    async componentDidMount() {
+        try {
+            const searchTagResponse = await fetch('/api/get_tags/');
+            const searchTags = await searchTagResponse.json();
+            const { tags, photographers } = searchTags;
+            // Sort by name and then by number, if the photographers have one
+            photographers.sort((a, b) => {
```

Comment:
Cleanup TODO (not preventing merge) -- looks to me like this sort can be done in the backend in the db query by Django using the `order_by` method of a queryset (https://docs.djangoproject.com/en/3.1/ref/models/querysets/#order-by). Much faster to do a sort this way if it's possible than doing it in the frontend like this.